File: tree.py

import statistics as st 


class Node():
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

class Tree():
    def __init__(self, value):
        self.root = Node(value)
        
    def insert(self, number:int):
        new_node = Node(number)
        current = self.root
        while(True):
            if current.value == number:
                return False
            elif new_node.value > number:
                current.left = new_node if current.left == None else current.left
                return True
            else: 
                current.right = new_node if current.right == None else current.right
                return True

if __name__ == '__main__':
    bst = Tree(6)
    bst.insert(8)
    print(bst.root.right.value)       
            
                
            

    

File: Tree.java
package DataStructures; 


public class Tree {
    static class Node{
        Node left, right; 
        int value; 
        public Node(int number){
            this.value = number;
            this.left = this.right = null; 
        }
    }

    private Node root; 
    private StringBuilder s = new StringBuilder(""); 

    public Tree(int number){
        this.root = new Node(number); 
    }

    public void insert(int number){
        Node newNode = new Node(number);
        this.insertRecursively(this.root, newNode);  
    }

    public Node insertRecursively(Node currentRoot, Node newNode){
        if(currentRoot == null){
            return newNode; 
        }else if(currentRoot.value > newNode.value){
            currentRoot.left = this.insertRecursively(currentRoot.left, newNode); 
        }else{
            currentRoot.right = this.insertRecursively(currentRoot.right, newNode); 
        }

        return currentRoot; 
    }

    public String inorder(){
        this.inorderTraversal(this.root);
        String p = new String(s); 
        this.s = new StringBuilder(" "); 
        return p; 
    }

    private void inorderTraversal(Node currentRoot){
        if(currentRoot == null){
           return; 
        }
        this.inorderTraversal(currentRoot.left);
        s.append(currentRoot.value).append(", "); 
        this.inorderTraversal(currentRoot.right);
    }

    public static void main(String[] args){
        Tree tr = new Tree(100); 
        tr.insert(49);
        tr.insert(53); 
        tr.insert(101); 
        tr.insert(79); 
        tr.insert(153); 
        tr.insert(125); 
        tr.insert(110); 
        String s = tr.inorder(); 
        System.out.println(s);
    }



}
